pkg/fleet-manager/fleet_plugin_distributedstorage.go

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright Kurator Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fleet
+
+import (
+	"context"
+	"time"
+
+	"helm.sh/helm/v3/pkg/kube"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	"kurator.dev/kurator/pkg/apis/fleet/v1alpha1"
+	"kurator.dev/kurator/pkg/fleet-manager/plugin"
+	"kurator.dev/kurator/pkg/infra/util"
+)
+
+func (f *FleetManager) reconcileDistributedStoragePlugin(ctx context.Context, fleet *v1alpha1.Fleet, fleetClusters map[ClusterKey]*fleetCluster) (kube.ResourceList, ctrl.Result, error) {
+	log := ctrl.LoggerFrom(ctx)
+
+	distributedStorageCfg := fleet.Spec.Plugin.DistributedStorage
+
+	if distributedStorageCfg == nil {
+		// reconcilePluginResources will delete all resources if plugin is nil
+		return nil, ctrl.Result{}, nil
+	}
+
+	fleetNN := types.NamespacedName{
+		Namespace: fleet.Namespace,
+		Name:      fleet.Name,
+	}
+
+	fleetOwnerRef := ownerReference(fleet)
+	var resources kube.ResourceList
+
+	// First install rook-operator for the specified multicluster.
+	for key, cluster := range fleetClusters {
+		b, err := plugin.RendeStorageOperator(f.Manifests, fleetNN, fleetOwnerRef, plugin.FleetCluster{
+			Name:       key.Name,
+			SecretName: cluster.Secret,
+			SecretKey:  cluster.SecretKey,
+		}, distributedStorageCfg)
+		if err != nil {
+			return nil, ctrl.Result{}, err
+		}
+
+		// apply Rook helm resources
+		rookResources, err := util.PatchResources(b)
+		if err != nil {
+			return nil, ctrl.Result{}, err
+		}
+		resources = append(resources, rookResources...)
+	}
+
+	log.V(4).Info("wait for rook helm release to be reconciled")
+	if !f.helmReleaseReady(ctx, fleet, resources) {
+		// wait for HelmRelease to be ready
+		return nil, ctrl.Result{
+			// HelmRelease check interval is 1m, so we set 30s here
+			RequeueAfter: 30 * time.Second,
+		}, nil
+	}
+
+	// After Rook operator are created, starts to install rook-ceph
+	if distributedStorageCfg.Storage == nil {
+		// reconcilePluginResources will delete all resources if plugin is nil
```

Comment:
cc @Xieql 

We donot support plugin uninstall now, right?

Reply:
we  support uninstall plugin. see https://kurator.dev/docs/fleet-manager/backup/backup-plugin/

```
1. Cleanup the Backup Plugin
If you only need to remove the backup plugin, simply edit the current fleet and remove the corresponding description:

kubectl edit fleet.fleet.kurator.dev quickstart
To check the results of the deletion, you can observe that the Velero components have been removed:

kubectl get po -A --kubeconfig=/root/.kube/kurator-member1.config
kubectl get po -A --kubeconfig=/root/.kube/kurator-member2.config
If you wish to reinstall the components later, you can simply edit the fleet and add the necessary configurations.

2. Cleanup the Fleet
When the fleet is deleted, all associated plugins will also be removed:

kubectl delete fleet.fleet.kurator.dev quickstart
```

Reply:
IIRC, The uninstall is supported by FluxCD. Ff `helmrelease` is removed,the uninstall will be proformed

Reply:
see https://github.com/kurator-dev/kurator/blob/52c2828d181fd377d3ba2fe520e0ca6eaeb73fe6/pkg/fleet-manager/fleet_plugin.go#L105-L107

Reply:
Thanks i find it